vowpalwabbit/cb_explore_adf_bag.cc

```diff
@@ -127,14 +127,14 @@ VW::LEARNER::base_learner* setup(VW::config::options_i& options, vw& all)
   new_options
       .add(make_option("cb_explore_adf", cb_explore_adf_option)
                .keep()
+               .necessary()
```

Comment:
cb_explore_adf is a weird one, in that `cb_explore_adf` is the reduction type; the other options marked as required is the explore type, which is a sub-option, only one of which can(?) be used (I'm not sure exactly what'll happen if multiple explore types are specified.. undefined behavior probably)

Reply:
Indeed cb_explore_adf is a weird one. If a user specifies more than one who knows what will happen, unless we test for it and add the safeguards to the reductions themselves. That could be a separate task unrelated to this refactor.